src/main/java/com/redhat/devtools/intellij/lsp4ij/LanguageServerExcecutionContext.java

```diff
@@ -0,0 +1,85 @@
+package com.redhat.devtools.intellij.lsp4ij;
+
+import com.intellij.openapi.util.Pair;
+import com.redhat.devtools.intellij.lsp4ij.LanguageServerWrapper;
+import com.redhat.devtools.intellij.lsp4ij.internal.CompletableFutureWrapper;
+import org.eclipse.lsp4j.CodeLens;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+import org.eclipse.lsp4j.services.LanguageServer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+public class LanguageServerExcecutionContext implements CancelChecker  {

```

Comment:
LanguageServerExecutionContext

Reply:
I did somerefactoring, now I rename this class with `CancellationSupport`